scripts/o3de/o3de/cmake.py

```diff
@@ -243,3 +242,120 @@ def get_enabled_gem_cmake_file(project_name: str = None,
+                                                             project_path=project_path, 
+                                                             include_manifest_gems=True, 
+                                                             include_engine_gems=True)
+
+    # First try to resolve with optional gems
+    results, errors = compatibility.resolve_gem_dependencies(gem_names_with_optional_gems, 
+                                                             all_gems_json_data, 
+                                                             engine_json_data, 
+                                                             include_optional=True)
+    if errors:
+        logger.warning('Failed to resolve dependencies with optional gems, trying without optional gems.')
+
+        # Try without optional gems
+        gem_names_without_optional = utils.get_gem_names_set(active_gem_names, include_optional=False)
+        results, errors = compatibility.resolve_gem_dependencies(gem_names_without_optional, 
+                                                                 all_gems_json_data, 
+                                                                 engine_json_data,
+                                                                 include_optional=False)
+
+    if errors:
+        logger.error(f'Failed to resolve dependencies:\n '+ 
+                    "\n  ".join(errors))
+        return 1
+
+    with resolved_gem_dependencies_output_path.open('w') as output:
+        # make a list of <gem_name>;<gem_path> for cmake
+        gem_paths = sorted(f"{gem.gem_json_data['gem_name'].strip()};{gem.gem_json_data['path'].resolve().as_posix()}" for _, gem in results.items())
+        # use dict to remove duplicates and preserve order so it's easier to read/debug
+        gem_paths = list(dict.fromkeys(gem_paths))
+        # join everything with a ';' character which is a list entry delimiter in CMake
+        # so the keys and values are all list entries
+        output.write(';'.join(gem_paths))
+
+    return 0
+
+def _resolve_gem_dependency_paths(args: argparse) -> int:
+    return resolve_gem_dependency_paths(
+                            engine_path=args.engine_path,
+                            project_path=args.project_path,
+                            resolved_gem_dependencies_output_path=args.gem_paths_output_file
+                             )
+
+def add_parser_args(parser):
+    group = parser.add_argument_group("resolve gem dependencies")
+    group.add_argument('-pp', '--project-path', type=pathlib.Path, required=False,
+                       help='The path to the project.')
+    group.add_argument('-ep', '--engine-path', type=pathlib.Path, required=False,
+                       help='The path to the engine.')
+    group.add_argument('-gpof', '--gem-paths-output-file', type=pathlib.Path, required=True,
+                       help='The path to the resolved gem paths output file.')
```

Comment:
This shouldn't be required.
If it is not set, then the file to open should be set to `sys.stdout`

Reply:
or... you can also add a `--dry-run` option to output to `sys.stdout`